subgraphs/uniswap-v2/src/common/helpers.ts

```diff
@@ -0,0 +1,518 @@
+    // side effects for other code paths.
+    mints.pop()
+    transaction.mints = mints
+    transaction.save()
+  }
+  burn.save()
+  // if accessing last one, replace it
+  if (burn.needsComplete) {
+    // TODO: Consider using .slice(0, -1).conc at() to protect against
+    // unintended side effects for other code paths.
+
+    if (burns != null) burns[burns.length - 1] = burn.id
+    else burns = [burn.id]
+  }
+  // else add new one
+  else {
+    if (burns == null) transaction.burns = [burn.id]
+    else transaction.burns = burns.concat([burn.id])
+  }
+  transaction.burns = burns
+  transaction.save()
+  pool.save()
+}
+
+// Generate the deposit entity and update deposit account for the according pool.
+export function createDeposit(event: ethereum.Event, amount0: BigInt, amount1: BigInt, sender: Address): void {
+  let transaction = getOrCreateTransaction(event)
+
+  let mints = transaction.mints!
+  let mint = getMint(mints[mints.length - 1])
+
+  let pool = getLiquidityPool(event.address.toHexString())
+
+  let token0 = getOrCreateToken(Address.fromString(pool.inputTokens[0]))
+  let token1 = getOrCreateToken(Address.fromString(pool.inputTokens[1]))
+  let tokenTracker0 = getOrCreateTokenTracker(Address.fromString(pool.inputTokens[0]))
+  let tokenTracker1 = getOrCreateTokenTracker(Address.fromString(pool.inputTokens[1]))
+
+  // update exchange info (except balances, sync will cover that)
+  let token0Amount = convertTokenToDecimal(amount0, token0.decimals)
+  let token1Amount = convertTokenToDecimal(amount1, token1.decimals)
+
+  let ether = getOrCreateEtherHelper()
+
+  // Gets token value in USD
+  let token0USD = tokenTracker0.derivedETH.times(ether.valueDecimal!)
+  let token1USD = tokenTracker1.derivedETH.times(ether.valueDecimal!)
+
+  let deposit = new Deposit(
+    event.transaction.hash
+      .toHexString()
+      .concat('-')
+      .concat(event.logIndex.toString())
+  )
+
+  deposit.hash = event.transaction.hash.toHexString()
+  deposit.logIndex = event.logIndex.toI32()
+  deposit.protocol = FACTORY_ADDRESS
+  deposit.to = mint.to.toHexString()
+  deposit.from = sender.toHexString()
+  deposit.blockNumber = event.block.number
+  deposit.timestamp = event.block.timestamp
+  deposit.inputTokens = [pool.inputTokens[0], pool.inputTokens[1]]
+  deposit.outputTokens = pool.outputToken
+  deposit.inputTokenAmounts = [amount0, amount1]
+  deposit.outputTokenAmount = mint.liquidity
+  deposit.amountUSD = token0USD.times(token0Amount).plus(token1USD.times(token1Amount))
+
+  mint.from = sender
+
+  UpdateDepositHelper(event.address)
+
+  deposit.save()
+  mint.save()
+}
+
+// Generate the withdraw entity
+export function createWithdraw(event: ethereum.Event, amount0: BigInt, amount1: BigInt, sender: Address, to: Address): void {
+  let transaction = getOrCreateTransaction(event)
+
+  let burns = transaction.burns!
+  let burn = getBurn(burns[burns.length - 1])
```

Comment:
Why are we only taking the last Burn in the transaction?

Reply:
Same as above